cmd/argot/tools/tools.go

```diff
@@ -127,37 +132,31 @@ func LoadConfig(configPath string) (*config.Config, error) {
 //
 // When args is not empty, only the target "" -> args is returned.
 // When the tool name is not recognized, all the targets in the config file are returned.
-func GetTargets(args []string, c *config.Config, tool config.ToolName) map[string][]string {
+func GetTargets(args []string, tag string, c *config.Config, tool config.ToolName) map[string][]string {
 	if len(args) > 0 {
 		return map[string][]string{"": args}
 	}
 	allTargets := c.GetTargetMap()
 	switch tool {
 	case config.TaintTool:
-		taintTargets := map[string][]string{}
-		for _, ttp := range c.TaintTrackingProblems {
-			for _, target := range ttp.Targets {
-				taintTargets[target] = allTargets[target]
-			}
-		}
-		return taintTargets
+		return targets(c.TaintTrackingProblems, allTargets, tag)
 	case config.BacktraceTool:
-		backtraceTargets := map[string][]string{}
-		for _, ttp := range c.SlicingProblems {
-			for _, target := range ttp.Targets {
-				backtraceTargets[target] = allTargets[target]
-			}
-		}
-		return backtraceTargets
+		return targets(c.SlicingProblems, allTargets, tag)
 	case config.SyntacticTool:
-		structInitTargets := map[string][]string{}
-		for _, sis := range c.SyntacticProblems.StructInitProblems {
-			for _, target := range sis.Targets {
-				structInitTargets[target] = allTargets[target]
-			}
-		}
-		return structInitTargets
+		return targets(c.SyntacticProblems.StructInitProblems, allTargets, tag)
 	default:
 		return allTargets
 	}
 }
+
+func targets[T config.TaggedSpec](problems []T, allTargets map[string][]string, tag string) map[string][]string {
```

Comment:
It seemed the most concise way for *slices* of interfaces, otherwise you need to convert explicitly slices of implementations to slices of interfaces by creating a new slice of interfaces.